tree-wide: Do not use fail to, error prefixes in error messages

mauriciovasquezbernal opened the pull request:
It's a bit annoying to tell people not to use "failed to, "fail", etc. prefixes in error messages when the code is plenty of them. Remove all of them in a single pass. 

See https://github.com/inspektor-gadget/inspektor-gadget/issues/881
Fixes https://github.com/inspektor-gadget/inspektor-gadget/issues/881

alban commented on cmd/common/utils/errors.go:
```diff
@@ -112,9 +112,9 @@ func WrapInErrInvalidArg(arg string, err error) error {
 
 func WrapInErrMarshalOutput(err error) error {
-	return fmt.Errorf("failed to marshal output: %w", err)
+	return fmt.Errorf("marshalling output: %w", err)
```
```suggestion
	return fmt.Errorf("marshaling output: %w", err)
```
Both are valid (American vs British English). I suggest to use a single `l` to be consistent with other errors

alban commented on cmd/kubectl-gadget/utils/trace.go:
```diff
@@ -318,7 +318,7 @@ func updateTraceOperation(trace *gadgetv1alpha1.Trace, operation string) error {
 	patchBytes, err := json.Marshal(patch)
 	if err != nil {
-		return fmt.Errorf("failed to marshal the operation annotations: %w", err)
+		return fmt.Errorf("marshalling the operation annotations: %w", err)
```
```suggestion
		return fmt.Errorf("marshaling the operation annotations: %w", err)
```

alban commented on cmd/kubectl-gadget/advise/network-policy.go:
```diff
@@ -135,17 +135,17 @@ func runNetworkPolicyReport(cmd *cobra.Command, args []string) error {
 	_, err = w.Write([]byte(adv.FormatPolicies()))
 	if err != nil {
-		return fmt.Errorf("failed to write file %q: %w", outputFileName, err)
+		return fmt.Errorf("writting file %q: %w", outputFileName, err)
```
```suggestion
		return fmt.Errorf("writing file %q: %w", outputFileName, err)
```

alban commented on pkg/gadgets/advise/seccomp/tracer/tracer.go:
```diff
@@ -81,7 +81,7 @@ func (t *Tracer) install() error {
 	})
 	if err != nil {
-		return fmt.Errorf("failed to open tracepoint: %w", err)
+		return fmt.Errorf("openning tracepoint: %w", err)
```
```suggestion
		return fmt.Errorf("opening tracepoint: %w", err)
```

alban replied:
or `attaching` to be consistent. Ditto in other occurrences.

alban commented on pkg/gadgets/internal/networktracer/tracer.go:
```diff
@@ -99,26 +99,26 @@ func newAttachment(
 	a.sockFd, err = rawsock.OpenRawSock(pid)
 	if err != nil {
-		return nil, fmt.Errorf("failed to open raw socket: %w", err)
+		return nil, fmt.Errorf("openning raw socket: %w", err)
```
```suggestion
		return nil, fmt.Errorf("opening raw socket: %w", err)
```

alban commented on pkg/gadgets/snapshot/process/tracer/tracer.go:
```diff
@@ -87,13 +87,13 @@ func runeBPFCollector(config *Config, enricher gadgets.DataEnricherByMntNs) ([]*
 	file, err := dumpTaskIter.Open()
 	if err != nil {
-		return nil, fmt.Errorf("failed to open BPF iterator: %w", err)
+		return nil, fmt.Errorf("openning BPF iterator: %w", err)
```
```suggestion
		return nil, fmt.Errorf("opening BPF iterator: %w", err)
```

alban commented on pkg/gadgets/snapshot/socket/tracer/tracer.go:
```diff
@@ -132,7 +132,7 @@ func (t *Tracer) RunCollector(pid uint32, podname, namespace, node string) ([]*s
 			reader, err := it.Open()
 			if err != nil {
-				return fmt.Errorf("failed to open BPF iterator: %w", err)
+				return fmt.Errorf("openning BPF iterator: %w", err)
```
```suggestion
				return fmt.Errorf("opening BPF iterator: %w", err)
```

alban commented on pkg/gadgets/top/ebpf/piditer/iter.go:
```diff
@@ -84,7 +84,7 @@ func NewTracer() (iter *PidIter, err error) {
 	rc, err := p.iter.Open()
 	if err != nil {
-		return nil, fmt.Errorf("failed to open iter: %w", err)
+		return nil, fmt.Errorf("openning iter: %w", err)
```
```suggestion
		return nil, fmt.Errorf("opening iter: %w", err)
```

alban commented on pkg/gadgets/trace/capabilities/tracer/tracer.go:
```diff
@@ -165,25 +165,25 @@ func (t *Tracer) install() error {
 	})
 	if err != nil {
-		return fmt.Errorf("failed to open tracepoint: %w", err)
+		return fmt.Errorf("openning tracepoint: %w", err)
```
```suggestion
		return fmt.Errorf("attaching tracepoint: %w", err)
```

alban commented on pkg/gadgets/trace/network/tracer/tracer.go:
```diff
@@ -152,19 +152,19 @@ func (t *Tracer) Attach(pid uint32) (err error) {
 
 	if a.sockFd, err = rawsock.OpenRawSock(pid); err != nil {
-		return fmt.Errorf("failed to open raw socket: %w", err)
+		return fmt.Errorf("openning raw socket: %w", err)
```
```suggestion
		return fmt.Errorf("opening raw socket: %w", err)
```

alban commented on pkg/gadgets/trace/tcpconnect/tracer/tracer.go:
```diff
@@ -110,39 +110,39 @@ func (t *Tracer) install() error {
 	t.v4EnterLink, err = link.Kprobe("tcp_v4_connect", t.objs.IgTcpcV4CoE, nil)
 	if err != nil {
-		return fmt.Errorf("error attaching program: %w", err)
+		return fmt.Errorf("attaching krpobe: %w", err)
```
```suggestion
		return fmt.Errorf("attaching kprobe: %w", err)
```

alban commented on pkg/gadgets/trace/tcpconnect/tracer/tracer.go:
```diff
@@ -110,39 +110,39 @@ func (t *Tracer) install() error {
 	t.v6EnterLink, err = link.Kprobe("tcp_v6_connect", t.objs.IgTcpcV6CoE, nil)
 	if err != nil {
-		return fmt.Errorf("error attaching program: %w", err)
+		return fmt.Errorf("attaching krpobe: %w", err)
```
```suggestion
		return fmt.Errorf("attaching kprobe: %w", err)
```

alban commented on pkg/gadgets/trace/tcpconnect/tracer/tracer.go:
```diff
@@ -110,39 +110,39 @@ func (t *Tracer) install() error {
 		t.v4ExitLink, err = link.Kretprobe("tcp_v4_connect", t.objs.IgTcpcV4CoX, nil)
 		if err != nil {
-			return fmt.Errorf("error attaching program: %w", err)
+			return fmt.Errorf("attaching kretrpobe: %w", err)
```
```suggestion
			return fmt.Errorf("attaching kretprobe: %w", err)
```

alban commented on pkg/gadgets/trace/tcpconnect/tracer/tracer.go:
```diff
@@ -110,39 +110,39 @@ func (t *Tracer) install() error {
 		t.v6ExitLink, err = link.Kretprobe("tcp_v6_connect", t.objs.IgTcpcV6CoX, nil)
 		if err != nil {
-			return fmt.Errorf("error attaching program: %w", err)
+			return fmt.Errorf("attaching kretrpobe: %w", err)
```
```suggestion
			return fmt.Errorf("attaching kretprobe: %w", err)
```

mauriciovasquezbernal commented:
> Thanks for the tedious work!
 

Thanks a lot for the detailed review and catching many stupid mistakes I did. 

> I started reviewing until pkg/gadgets/top/file/tracer/tracer.go
> 
> The unit test needs to be fixed too.

What do you mean for unit tests? I think it's fine to use "failed to", "error while", in `t.Errorf` given that it's not concatenated to other errors. WDYT?

mauriciovasquezbernal commented:
> Did you use a script for this refactoring? So, we can use a script to check your script?

I did it by hand. A lot of search and replace. 

> Is there something we can use to prevent ourselves adding back "fail to" and others?
@invidian proposed something in https://github.com/inspektor-gadget/inspektor-gadget/issues/881#issuecomment-1231299788. I think we can add to Github actions. I'll look into it.

alban commented:
> What do you mean for unit tests?

I meant this:
```
=== RUN   TestPrintTraceFeedback
    trace_test.go:99: 'MyPrefix: running gadget on node "node": Err/Warn Message
        ' != 'MyPrefix: failed to run gadget on node "node": Err/Warn Message
        '
--- FAIL: TestPrintTraceFeedback (0.00s)
```

Since your PR changed this line:
```
cmd/common/utils/errors.go:71:  return fmt.Errorf("failed to run gadget on node %q: %w", node, err)
```

The following test is no longer passing:
```
        expected := "MyPrefix: failed to run gadget on node \"node\": Err/Warn Message\n"
```

> I think it's fine to use "failed to", "error while", in t.Errorf given that it's not concatenated to other errors. WDYT?

Yes, totally fine :)

mauriciovasquezbernal commented:
> @invidian proposed something in [#881 (comment)](https://github.com/inspektor-gadget/inspektor-gadget/issues/881#issuecomment-1231299788). I think we can add to Github actions. I'll look into it.

Added it.

> I meant this:

Fixed.

invidian commented on semgrep_rules.yaml:
```diff
@@ -0,0 +1,7 @@
+rules:
```
Maybe we can make it a `.semgrep_rules.yaml` to avoid polluting repo's root dir?

mauriciovasquezbernal replied:
Makes sense. Thanks!

invidian reviewed (commented):
Awesome work @mauriciovasquezbernal. And cool that you decided to adopt my rule set :)

mauriciovasquezbernal commented on pkg/gadgets/top/ebpf/piditer/iter.go:
```diff
@@ -56,24 +56,24 @@ func NewTracer() (iter *PidIter, err error) {
 
 	if err := kallsyms.SpecUpdateAddresses(spec, []string{"bpf_prog_fops"}); err != nil {
-		return nil, fmt.Errorf("error RewriteConstants: %w", err)
+		return nil, fmt.Errorf("rewriting constants: %w", err)
```
Updated.

mauriciovasquezbernal commented:
Thanks for the review folks!
